### PR TITLE
Bug 2047308:Remove metrics and events for master port offsets

### PR DESCRIPTION
**Pull request by aneeshkp**
Signed-off-by: Aneesh Puttur <aneeshputtur@gmail.com>
This PR will reduce events and metrics to  master and clock_realtime only
``` 
# TYPE openshift_ptp_frequency_adjustment_ns gauge
openshift_ptp_frequency_adjustment_ns{from="master",iface="ens7fx",node="cnfde7.ptp.lab.eng.bos.redhat.com",process="ptp4l"} -4102
openshift_ptp_frequency_adjustment_ns{from="phc",iface="CLOCK_REALTIME",node="cnfde7.ptp.lab.eng.bos.redhat.com",process="phc2sys"} -79997
# HELP openshift_ptp_interface_role 0 = PASSIVE, 1 = SLAVE, 2 = MASTER, 3 = FAULTY, 4 =  UNKNOWN
# TYPE openshift_ptp_interface_role gauge
openshift_ptp_interface_role{iface="ens7f0",node="cnfde7.ptp.lab.eng.bos.redhat.com",process="ptp4l"} 2
openshift_ptp_interface_role{iface="ens7f1",node="cnfde7.ptp.lab.eng.bos.redhat.com",process="ptp4l"} 1


 TYPE openshift_ptp_threshold gauge
openshift_ptp_threshold{node="cnfde7.ptp.lab.eng.bos.redhat.com",profile="profile1",threshold="HoldOverTimeout"} 65
openshift_ptp_threshold{node="cnfde7.ptp.lab.eng.bos.redhat.com",profile="profile1",threshold="MaxOffsetThreshold"} 3000
openshift_ptp_threshold{node="cnfde7.ptp.lab.eng.bos.redhat.com",profile="profile1",threshold="MinOffsetThreshold"} -3000
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,8 +5,8 @@
 VERSION ?=latest
 # Default image tag
 
-SIDECAR_IMG ?= quay.io/aneeshkp/cloud-event-proxy:$(VERSION)
-CONSUMER_IMG ?= quay.io/aneeshkp/cloud-native-event-consumer:$(VERSION)
+IMG ?= quay.io/redhat-cne/cloud-event-proxy:$(VERSION)
```
Comment (jzding) [inline Makefile:8]: Use openshift image? `quay.io/openshift/origin-cloud-event-proxy`

File: docs/development.md [hunk 2]
```diff
@@ -38,8 +38,8 @@ make run-consumer
 ### Push images to a repo
 
 ```shell
-podman push localhost/cloud-event-proxy:${TAG} quay.io/aneeshkp/cloud-event-proxy:latest
-podman push localhost/cloud-native-event-consumer:${TAG}quay.io/aneeshkp/cloud-native-event-consumer:latest
+podman push localhost/cloud-event-proxy:${TAG} quay.io/redhat-cne/cloud-event-proxy:latest
+podman push localhost/cloud-event-consumer:${TAG}quay.io/redhat-cne/cloud-event-consumer:latest
```
Comment (jzding) [inline docs/development.md:42]: add space after tag: `${TAG}quay.io`

File: examples/manifests/kustomization.yaml [hunk 3]
```diff
@@ -8,11 +8,11 @@ resources:
 - consumer.yaml
 replicas:
 - count: 1
-  name: cloud-native-consumer-deployment
+  name: cloud-consumer-deployment
 images:
 - name: cloud-event-proxy
-  newName: quay.io/aneeshkp/cloud-event-proxy
+  newName: quay.io/redhat-cne/cloud-event-proxy
```
Comment (jzding) [inline examples/manifests/kustomization.yaml:14]: quay.io/openshift/origin-cloud-event-proxy

File: hack/deploy_test.sh [hunk 4]
```diff
@@ -8,8 +8,8 @@ source "$DIR/common/consumer.sh"
 source "$DIR/common/router.sh"
 
 KUBECONFIG="${KUBECONFIG:-}"
-CNE_IMG="${CNE_IMG:-quay.io/aneeshkp/cloud-event-proxy}"
-CONSUMER_IMG="${CONSUMER_IMG:-quay.io/aneeshkp/cloud-native-event-consumer}"
+CNE_IMG="${CNE_IMG:-quay.io/redhat-cne/cloud-event-proxy}"
```
Comment (jzding) [inline hack/deploy_test.sh:11]: quay.io/openshift/origin-cloud-event-proxy

File: plugins/ptp_operator/config/config.go [hunk 5]
```diff
@@ -196,6 +198,16 @@ func (l *LinuxPTPConfigMapUpdate) UpdatePTPThreshold() {
...
+		l.EventThreshold[*profile.Name] = &PtpClockThreshold{
+			HoldOverTimeout:    holdOverTh,
+			MaxOffsetThreshold: maxOffsetTh,
+			MinOffsetThreshold: minOffsetTh,
+			Close:              make(chan struct{}),
+		}
+		log.Infof("update ptp threshold values for %s\n holdoverTimeout: %d\n maxThreshold: %d\n minThreshold: %d\n",
+			*profile.Name, holdOverTh, maxOffsetTh, minOffsetTh)
+
+		/* move threshold per profile
```
Comment (jzding) [inline plugins/ptp_operator/config/config.go:210]: delete these if no longer needed?

File: Makefile (hunk 1 above)
Comment (aneeshkp) [reply Makefile:8]: this for developer, you can't build and push it to this locally

File: docs/development.md (hunk 2 above)
Comment (aneeshkp) [reply docs/development.md:42]: +1

File: examples/manifests/kustomization.yaml (hunk 3 above)
Comment (aneeshkp) [reply examples/manifests/kustomization.yaml:14]: done

File: Makefile (hunk 1 above)
Comment (aneeshkp) [reply Makefile:8]: alright changed it to openshift, anyway user can override it with $IMG

File: plugins/ptp_operator/config/config.go (hunk 5 above)
Comment (aneeshkp) [reply plugins/ptp_operator/config/config.go:210]: Done

File: hack/deploy_test.sh (hunk 4 above)
Comment (aneeshkp) [reply hack/deploy_test.sh:11]: Done

File: plugins/ptp_operator/ptp_operator_plugin.go [hunk 6]
```diff
@@ -48,6 +46,10 @@ const (
 	ptpConfigDir       = "/var/run/"
 	phc2sysProcessName = "phc2sys"
 	ptp4lProcessName   = "ptp4l"
+	//ClockRealTime is the slave
+	ClockRealTime = "CLOCK_REALTIME"
+	//MasterClockType is teh slave sync slave clock to master
```
Comment (jzding) [inline plugins/ptp_operator/ptp_operator_plugin.go:51]: teh -> the

File: plugins/ptp_operator/metrics/metrics.go [hunk 7]
```diff
@@ -308,11 +308,16 @@ func (p *PTPEventManager) ExtractMetrics(msg string) {
...
-			UpdatePTPMetrics(master, processName, interfaceName, ptpOffset, maxPtpOffset, frequencyAdjustment, delay)
-		case "":
-			//do nothing
-		default:
-			UpdatePTPMetrics(phc, processName, interfaceName, ptpOffset, maxPtpOffset, frequencyAdjustment, delay)
+			ptpInterface, _ = ptp4lCfg.ByRole(types.SLAVE)
+			var alias string
+			if ptpInterface.Name != "" {
+				r := []rune(ptpInterface.Name)
+				alias = string(r[:len(r)-1]) + "x"
```
Comment (jzding) [inline plugins/ptp_operator/metrics/metrics.go:315]: is it possible to have a double digits index number a the end?

Comment (aneeshkp) [reply plugins/ptp_operator/metrics/metrics.go:315]: This is for network card index so one card won't have 10 ports.

File: plugins/ptp_operator/ptp_operator_plugin.go (hunk 6 above)
Comment (aneeshkp) [reply plugins/ptp_operator/ptp_operator_plugin.go:51]: done

**Review by jzding — APPROVED** [review]

**Pull request merged**